internal/kubeclient/indexer.go

```diff
@@ -87,18 +87,27 @@ func indexStagesByAnalysisRun(shardName string) client.IndexerFunc {
 		}
 
 		stage := obj.(*kargoapi.Stage) // nolint: forcetypeassert
-		if stage.Status.CurrentFreight == nil ||
-			stage.Status.CurrentFreight.VerificationInfo == nil ||
-			stage.Status.CurrentFreight.VerificationInfo.AnalysisRun == nil {
+
+		current := stage.Status.FreightHistory.Current()
+		if current == nil || len(current.Freight) == 0 {
 			return nil
 		}
-		return []string{
-			fmt.Sprintf(
-				"%s:%s",
-				stage.Status.CurrentFreight.VerificationInfo.AnalysisRun.Namespace,
-				stage.Status.CurrentFreight.VerificationInfo.AnalysisRun.Name,
-			),
+
+		var analysisRuns []string
```

Comment:
This requires more research, as it is allowed (see https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.18.4/pkg/client#FieldIndexer) but has indications it may not work nicely in every scenario.